### ISSUE TITLE
The `fixtures/{first,}/*` pattern returns incorrect results

### ISSUE THREAD

**Issue by mrmlnc** [mrmlnc/fast-glob#394]
### Environment

  * OS Version: macOS 13.1
  * Node.js Version: v16.16.0

### Actual behavior

```js
{
  entries: [
    'fixtures/first/file.md',
    'fixtures/first/nested'
  ]
}
```

### Expected behavior

The `fixtures/{first,}/*` can be considered as two patterns:

* `fixtures/first/*`
* `fixtures/*`

```js
{
  entries: [
    'fixtures/third',
    'fixtures/second',
    'fixtures/file.md',
    'fixtures/first',
    'fixtures/first/nested',
    'fixtures/first/file.md'
  ]
}
```

### Code sample

```js
const incorrect = fg.sync(['fixtures/{first,}/*'], {
	onlyFiles: false,
});

const correct = fg.sync(['fixtures/first/*', 'fixtures/*'], {
	onlyFiles: false,
});

console.dir({ incorrect, correct }, { colors: true, compact: false, depth: null });

```

**Comment by mrmlnc** [comment]
The `micromatch` package generates the following regexp: `^(?:fixtures\/(first|)\/(?!\.)(?=.)[^/]*?\/?)$`.

**Comment by mrmlnc** [comment]
Already fixed in the master by #395 in this package. Will be shipped with [`3.3.0`](https://github.com/mrmlnc/fast-glob/issues/396).
